lib/rules/prop-types.js

```diff
@@ -606,6 +606,10 @@ module.exports = {
           type = 'destructuring';
           properties = node.params[0].properties;
           break;
+        case 'MethodDefinition':
+          type = 'destructuring';
+          properties = node.value.params[0].properties;
```

Comment:
is there any chance a MethodDefinition could have a `node` without a `value`, or a `node.value` without a `params`, or a `node.value.params` without a `0th` item?

it'd be ideal if this was more defensive (like you're already doing below), so the rule doesn't crash.